src/main/scala/com/ibm/developer/eventstore/akka/OnlineRetailWebServer.scala

```diff
@@ -87,45 +98,26 @@ class OnlineRetailWebServer extends HttpApp {
     }
   }
 
-  override protected def routes: Route =
-    pathSingleSlash {
-      complete {
-        println("GET /")
-        HttpEntity(
-          ContentTypes.`text/html(UTF-8)`,
-          "<html><body>OnlineRetailWebServer is running!</body></html>")
-      }
-    } ~
-      pathPrefix("websocket") {
-        path("orderitem") {
-          println("Incoming ws: websocket/orderitem")
-          handleWebSocketMessages(websocket)
-        }
-      }
-
-  override protected def postHttpBindingFailure(cause: Throwable): Unit =
-    println(s"postHttpBindingFailure: $cause")
-
-  def start(host: String = "localhost", port: Int = 8080): Future[Done] = {
-    val settings = ServerSettings(system.settings.config)
-    Future {
-      startServer(host, port, settings, system)
-    }.map(_ => Done)
-  }
-
-  override protected def waitForShutdownSignal(system: ActorSystem)(implicit ec: ExecutionContext): Future[Done] =
-    shutdownPromise.future
-
-  def stop(): Future[Terminated] = {
-    shutdownPromise.tryComplete(Success(Done))
-    system.terminate()
-  }
+  def routes: Route =
+    extractLog { logger =>
+      concat(
```

Comment:
We recommend `concat` over `~` as it is easy to miss out on a dangling `~` or break it with a linefeed too many, while `concat(subrouteA, subRouteB)` can't go wrong.